diff --git a/alpha/template.php b/alpha/template.php
index 6dda153..0b0a210 100644
--- a/alpha/template.php
+++ b/alpha/template.php
@@ -196,12 +196,16 @@ function alpha_page_alter(&$vars) {
     }
   }
 
-  if (!module_implements('alpha_page_structure_alter')) {
-    alpha_alter('alpha_page_structure', $vars, $theme->theme);
-  }
-  else {
-    drupal_alter('alpha_page_structure', $vars, $theme->theme);
+  $executed = &drupal_static(__FUNCTION__, FALSE);
+  if (!$executed) {
+    if (!module_implements('alpha_page_structure_alter')) {
+      alpha_alter('alpha_page_structure', $vars, $theme->theme);
+    }
+    else {
+      drupal_alter('alpha_page_structure', $vars, $theme->theme);
+    }
   }
+  $executed = TRUE;
 }
 
 /**
